cli/types/index.d.ts

```diff
@@ -910,6 +927,165 @@ declare namespace Cypress {
     cancable?: boolean;
   }
 
+  interface ConfigOptions {

```

Comment:
so we allow all options to be optional using `Partial<ConfigOptions>`? that is pretty slick if true!

Reply:
Yes. `Partial` is a mapped type that iterates over the keys and makes each optional. It is really useful if you want an interface without optional parameters and make them optional for things like default overrides.

Mapped types are pretty cool. TypeScript `lib.d.ts` defines a few globally: https://github.com/Microsoft/TypeScript/blob/master/lib/lib.d.ts#L1329